include/deal.II/base/tensor.h

```diff
@@ -1293,7 +1293,7 @@ constexpr DEAL_II_ALWAYS_INLINE
   AssertIndexRange(i, dim);
 #  endif
 
-  return values[i];
+  return values[i < dim ? i : 0];
```

Comment:
This seems like the wrong approach. For reference, the warning is this:
```
/.../dealii/include/deal.II/base/tensor.h:1296:16: warning: array subscript 2 is above array bounds of ‘const dealii::Tensor<0, 2, double> [2]’ [-Warray-bounds]
```
So we have a rank-0 tensor (a scalar), for which the only valid index is `i=0`. As a matter of fact, I don't understand why for rank=0 we use this operator to begin with -- we have a specialization `Tensor<0,dim,Number>` for this.